signing/src/main/java/tech/pegasys/web3signer/signing/secp256k1/SingleSignerProvider.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2023 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.signing.secp256k1;
+
+import java.security.interfaces.ECPublicKey;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+
+public class SingleSignerProvider implements SignerProvider {
```

Comment:
This should be moved to the integration test package as this is only needed for the integration test now, we don't have a single provider mode in web3signer.

Reply:
moved